### PR TITLE
Speed up co-occurrence matrix generation

### PR DESCRIPTION
**Pull request by rootAvish**
There was a lot of code that was basically going around in a circle to achieve a very straightforward thing(thanks to pandas) and doubling the time required for co-occurrence matrix generation.

This fixes that and cleans up some code elsewhere.

~~Please note there's a failing unit test, that's due to change in method signature and will be fixed shortly.~~

### CODE REVIEW

File: analytics_platform/kronos/softnet/src/kronos_dependency_generator.py [hunk 1]
```diff
@@ -103,19 +103,12 @@ def _generate_component_class_to_package_edge_list_and_dict(cls, package_list,
...
-                        component_class] = temp_package_list
-                else:
-                    component_class_to_package_dict[component_class] = [package]
-
+                component_class_to_package_edge_list.append({
+                    softnet_constants.EDGE_DICT_FROM: component_class,
+                    softnet_constants.EDGE_DICT_TO: package
+                })
+                component_class_to_package_dict[component_class] = \
+                    component_class_to_package_dict.get(component_class, []) + [package]
```
Comment (miteshvp) [inline analytics_platform/kronos/softnet/src/kronos_dependency_generator.py]: Nice

File: analytics_platform/kronos/softnet/src/cooccurrence_matrix_generator.py [hunk 2]
```diff
@@ -71,31 +75,21 @@ def get_intent_occurrence(cls, row_intent_component_dict):
...
-                    component_class]
-                row_component_package_df = cooccurrence_matrix.loc[[row_id],
-                                                                   component_package_list]
-                row_component_package_dict = row_component_package_df.to_dict(
-                    orient="index")[row_id]
-                component_class_occurrence = cls.get_component_class_occurrence(
-                    row_component_package_dict)
-                cooccurrence_matrix.loc[
-                    [row_id], component_class] = component_class_occurrence
+                cooccurrence_matrix.loc[[row_id], package_topic_map.get(package)] = 1
```
Comment (miteshvp) [inline analytics_platform/kronos/softnet/src/cooccurrence_matrix_generator.py]: This is definitely much cleaner approach. But if we want to build the tree (more layers) then we may think on similar lines on `get_intent_occurrence`. LGTM in its current form.

File: analytics_platform/kronos/softnet/src/offline_training.py [hunk 3]
```diff
@@ -89,6 +89,13 @@ def generate_and_save_cooccurrence_matrices(input_kronos_dependency_data_store,
         input_kronos_dependency_data_store=input_kronos_dependency_data_store,
         additional_path=additional_path)
 
+    package_topic_json = \
+        input_kronos_dependency_data_store.read_json_file(os.path.join(
+                                                      additional_path,
+                                                      softnet_constants.GNOSIS_PTM_OUTPUT_PATH))
+    package_topic_dict = dict(package_topic_json)
```
Comment (miteshvp) [inline analytics_platform/kronos/softnet/src/offline_training.py]: Do you want to consider lowercase here?

**Comment by miteshvp** [comment]
@rootAvish - just one minor nitpick. Else LGTM.

File: analytics_platform/kronos/softnet/src/offline_training.py (hunk 3 above)
Comment (rootAvish) [reply analytics_platform/kronos/softnet/src/offline_training.py]: That is already taken care of - this package topic dict is the output of the first Gnosis stage (see the line above), so it contains the "c_xx" tags and all lowercase package names.

**Pull request merged**